File: BehaviorPattern/Visitor/Java/Concretes/ConcreteVisitor2.java
package Concretes;

import Interfaces.Visitor;

public class ConcreteVisitor2 implements Visitor {
    @Override
    public void visit(ConcreteElementA element) {
        System.out.println("ConcreteVisitor2: " + element.operationA());
    }

    @Override
    public void visit(ConcreteElementB element) {
        System.out.println("ConcreteVisitor2: " + element.operationB());
    }
}

File: BehaviorPattern/Visitor/Golang/Concretes/concrete_visitor_2.go
package Concretes

import (
    "fmt"
    "visitorpattern/Interfaces"
)

// ConcreteVisitor2 is a struct that implements the Visitor interface
type ConcreteVisitor2 struct{}

// Visit implements the visit method for Elements
func (v *ConcreteVisitor2) Visit(element Interfaces.Element) {
    switch e := element.(type) {
    case *ConcreteElementA:
        fmt.Println("ConcreteVisitor2:", e.OperationA())
    case *ConcreteElementB:
        fmt.Println("ConcreteVisitor2:", e.OperationB())
    }
}
